src/ts/scripts/compare-bench.ts

```diff
@@ -0,0 +1,187 @@
+import { writeFileSync, readFileSync } from 'fs';
+import { resolve } from 'path';
+
+interface CircuitData {
+  name: string;
+  totalGateCount: number;
+  gateCounts: Array<{
+    circuitName: string;
+    gateCount: number;
+  }>;
+  gas: {
+    gasLimits: {
+      daGas: number;
+      l2Gas: number;
+    };
+    teardownGasLimits: {
+      daGas: number;
+      l2Gas: number;
+    };
+  };
+}
+
+interface GateCounts {
+  summary: Record<string, number>;
+  results: CircuitData[];
+  gasSummary: Record<string, number>;
+}
+
+interface MetricComparison {
+  main: number;
+  pr: number;
+  diff: number;
+}
+
+interface ComparisonResult {
+  gates: MetricComparison;
+  daGas: MetricComparison;
+  l2Gas: MetricComparison;
+}
+
+const formatDiff = (mainValue: number, prValue: number): string => {
+  const diff = mainValue - prValue;
```

Comment:
```suggestion
  const diff = prValue - mainValue;
```
I think it makes sense read the report's diff as: how much cheaper or more expensive the PR is with respect to main, not the other way around.

Also note that the `+100%` and `-100%` cases are inverted with respect to `diff`